### ISSUE TITLE
what is the face feature if the face detector do not detect any face in the images? 

### ISSUE THREAD

**Issue by LiuJoffrey** [hqqasw/person-search-PPCC#6]
Hi,
I would like to reproduce your work with my own dataset. First, I need to use the face detector to get the location of the face from cast or tracklet image, and if it can't find any face from the image, what is the face feature for that tracklet?

Now I am using the Restnet50 and Resnet101 to extract the body and face feature, and then using the linear layer to downsample the features to 256 dimensions, is that correct? Then, I would use the feature to calculate the cosine similarity for both face to trcaklet and tracklet to tracklet. Finally,  using the similarity score to run your code, is that the correct way to reproduce the work. Thank you so much.

**Comment by LiuJoffrey** [comment]
I also want to know how to train the feature extractor model, is that using the common classifier model with crossentropy loss to train or using the cast and candidate image to calculate cosine similarity to train the model(if they are the same person, the label is 1, otherwise the label is -1) with mean square error loss? Thank you so much for helping me.

**Comment by hqqasw** [comment]
1. If no face detected, the feature would be set as 0.
2. The face feature is extracted by a Resnet-101 trained on Celeb-1M and the body feature is extracted by a Resnet-50 trained on CSM.
3. The face model and the body model are all trained with cross-entropy loss.

**Comment by LiuJoffrey** [comment]
Ok, thanks for your reply. I will try it to train the feature extractor.
